packages/codemirror-blocks/src/components/Node.tsx

```diff
@@ -229,6 +229,17 @@ const Node = ({ expandable = true, ...props }: Props) => {
     }),
   });
 
+  const nodeElementRef = useRef<HTMLSpanElement>(null);
+
+  useEffect(() => {
+    if (nodeElementRef.current) {
+      nodeElementMap.set(props.node.id, nodeElementRef.current);
+      return () => {
+        nodeElementMap.delete(props.node.id);
+      };
+    }
+  }, [props.node.id]);
+
```

Comment:
Making sure I really understand what's happening in this hook....

Once the ref is set (on render), the DOM node is stored in the `nodeElementMap` alongside the `id` key. Then a cleanup function is returned - which won't be called until the component is unmounted - which deletes the key/value pair from the map.

Do I have this right?

Reply:
yep, that's correct.